--- api/Saxon.Api/Model.cs	2007-06-12 11:56:08.000000000 +0000
+++ api/Saxon.Api/Model.new.cs	2007-06-12 11:54:47.000000000 +0000
@@ -14,7 +14,7 @@
 using JAtomicType = net.sf.saxon.type.AtomicType;
 using JSchemaType = net.sf.saxon.type.SchemaType;
 using JType = net.sf.saxon.type.Type;
-
+using net.sf.saxon.dotnet;
 
 namespace Saxon.Api {
 
@@ -334,7 +334,10 @@
             }
             this.value = result;
         }
-
+	/// insert from Dr. Kay
+	public static XdmAtomicValue wrapExternalObject(object external) {
+            return (XdmAtomicValue)XdmValue.Wrap(new DotNetObjectValue(external));
+        }
 
         /// <summary>
         /// Convert the atomic value to a string
@@ -1189,4 +1192,4 @@
 // Portions created by (your name) are Copyright (C) (your legal entity). All Rights Reserved.
 //
 // Contributor(s): none.
-//
\ No newline at end of file
+//
