--- Saxon/net/sf/saxon/number/Numberer_de.java	2007-06-12 11:34:12.000000000 +0000
+++ Saxon/net/sf/saxon/number/Numberer_de.new.java	2007-06-12 10:37:22.000000000 +0000
@@ -60,10 +60,7 @@
         }
     }
 
-    private static String[] germanUnits = {
-        "", "Eins", "Zwei", "Drei", "Vier", "Fünf", "Sechs", "Sieben", "Acht", "Neun",
-        "Zehn", "Elf", "Zwölf", "Dreizehn", "Vierzehn", "Fünfzehn", "Sechszehn",
-        "Siebzehn", "Achtzehn", "Neunzehn"};
+    private static String[] germanUnits = {"", "Eins", "Zwei", "Drei", "Vier", "Funf", "Sechs", "Sieben", "Acht", "Neun", "Zehn", "Elf", "Zwolf", "Dreizehn", "Vierzehn", "Fünfzehn", "Sechszehn", "Siebzehn", "Achtzehn", "Neunzehn"};
 
     private static String[] germanTens = {
         "", "Zehn", "Zwanzig", "Dreißig", "Vierzig", "Fünfzig",
